yapapi/network.py

```diff
@@ -54,6 +55,26 @@ def get_websocket_uri(self, port: int) -> str:
         return f"{net_api_ws}/net/{self.network.network_id}/tcp/{self.ip}/{port}"
 
 
+class NetworkStateMachine(StateMachine):
+    """State machine describing the states and lifecycle of a :class:`Network` instance."""
+
+    # states
+    initialized = State("initialized", initial=True)
+    creating = State("creating")
+    ready = State("ready")
+    removing = State("removing")
+    removed = State("removed")
+
+    # transitions
+    add_address = creating.to.itself() | ready.to.itself()
+    add_node = ready.to.itself()
+    get_id = creating.to.itself() | ready.to.itself() | removing.to.itself()
+
+    # lifecycle
+    create = initialized.to(creating) | creating.to(ready)
+    remove = ready.to(removing) | removing.to(removed)
```

Comment:
and actually, I'd do the same with `remove` -> have e.g. `stop which transitions from `ready` to `removing` and `remove` that moves from `removing` to `removed` ...

Reply:
Updated in b319b73